### PR TITLE
Pal 538 k8s shared storage for jars

### PR DESCRIPTION
**Pull request by dev-958**
Requires integration with PAL-683.

### CODE REVIEW

File: NOTICES.md [hunk 1]
```diff
@@ -22,9 +22,9 @@ List of third-party dependencies grouped by their license type
...
 * Spring Cloud Starter Netflix Eureka Server ([org.springframework.cloud:spring-cloud-starter-netflix-eureka-server:2.2.2.RELEASE](https://projects.spring.io/spring-cloud))
 * Spring Cloud Starter OpenFeign ([org.springframework.cloud:spring-cloud-starter-openfeign:2.2.2.RELEASE](https://projects.spring.io/spring-cloud))
-* common ([uk.gov.gchq.palisade:common:0.4.0-SNAPSHOT](https://github.com/gchq/Palisade-common))
-* hadoop-reader ([uk.gov.gchq.palisade:hadoop-reader:0.4.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-reader))
-* hadoop-resource ([uk.gov.gchq.palisade:hadoop-resource:0.4.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-resource))
+* common ([uk.gov.gchq.palisade:common:0.4.0-20200622.103134-73](https://github.com/gchq/Palisade-common))
+* hadoop-reader ([uk.gov.gchq.palisade:hadoop-reader:0.4.0-20200622.103205-31](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-reader))
+* hadoop-resource ([uk.gov.gchq.palisade:hadoop-resource:0.4.0-20200622.103134-19](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-resource))
```
Comment (ac74475) [inline NOTICES.md]: revert these changes

File: templates/redis-pv.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+{{- if .Values.redis.install }}
 apiVersion: v1
 kind: PersistentVolume
 metadata:
```
Comment (ac74475) [inline templates/redis-pv.yaml]: Are we sure we want persistentVolumeReclaimPolicy: Retain rather than delete. Given we changed this for the other PVs

File: templates/kafka-pv.yaml [hunk 3]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+{{- if .Values.kafka.install }}
 apiVersion: v1
 kind: PersistentVolume
 metadata:
```
Comment (ac74475) [inline templates/kafka-pv.yaml:18]: Are we sure we want persistentVolumeReclaimPolicy: Retain rather than delete. Given we changed this for the other PVs

**Review by ac74475 — COMMENTED** [review]
a small change and a couple of questions. Also would be worth using indentation on the _helper.tfl files if it is possible to improve there readability

File: templates/redis-pv.yaml (hunk 2 above)
Comment (dev-958) [reply templates/redis-pv.yaml]: This will get changed on merge from develop.

File: templates/kafka-pv.yaml (hunk 3 above)
Comment (dev-958) [reply templates/kafka-pv.yaml:18]: ditto.

**Comment by dev-958** [comment]
> a small change and a couple of questions. Also would be worth using indentation on the _helper.tfl files if it is possible to improve there readability

I try to strike a balance so that the target template retains it's correct tab flow so errors are easily visible. Not always that easy though.

File: charts/palisade-service/templates/_helpers.tpl [hunk 4]
```diff
@@ -70,3 +70,30 @@ Determine ingress root url
...
+{{- define "palisade-service.deployment.name" }}
+{{- include "palisade-service.deployment.path" . | base }}
+{{- end }}
+
+{{/*
+Calculate a storage full name based on the code release artifact id or the supplied value of codeRelease
+*/}}
+{{- define "palisade-service.deployment.fullname" }}
+{{- .Values.global.persistence.classpathJars.name }}-{{- include "palisade-service.deployment.name" . }}
+{{- end }}
```
Comment (nw1984) [inline charts/palisade-service/templates/_helpers.tpl]: Add newline at EOF for windows compatability

File: data-service/src/main/resources/values.yaml [hunk 5]
```diff
@@ -31,6 +32,12 @@ resources:
...
 
+autoscaling:
+  targetCPUUtilizationPercentage: 50
+  targetMemoryUtilizationPercentage: 50
+  minReplicas: 1
+  maxReplicas: 3
+
 nodeSelector: {}
 tolerations: []
 affinity: {}
```
Comment (nw1984) [inline data-service/src/main/resources/values.yaml]: ditto CRLF

File: templates/classpath-jars-sc.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+{{- if eq .Values.hosting "local" }}
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: {{ .Values.global.persistence.classpathJars.name }}
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+{{- end }}
```
Comment (nw1984) [inline templates/classpath-jars-sc.yaml]: A question - The SC on AWS needs to bind to the aws efs csi driver in order to mount the volume. This SC is only for local. On develop we have storageclass.yaml - does it make sense to merge this SC and the AWS version in storageclass.yaml?

File: templates/kafka-pv.yaml [hunk 7]
```diff
@@ -38,3 +39,4 @@ spec:
     volumeAttributes:
       path: {{ .Values.global.persistence.kafka.aws.volumePath }}
   {{- end }}
+{{- end }}
```
Comment (nw1984) [inline templates/kafka-pv.yaml]: CRLF

**Review by nw1984 — DISMISSED** [review]

File: NOTICES.md (hunk 1 above)
Comment (dev-958) [reply NOTICES.md]: done

**Review by nw1984 — DISMISSED** [review]

File: values.yaml [hunk 8]
```diff
@@ -60,15 +60,23 @@ global:
...
+
   # see charts/kafka/charts/zookeeper
   zookeeperConnectString: palisade-zookeeper:2181
 
   # persistent-volume and associated claims configurations
   persistence:
-    storageClassDeploy: false
-    storageClassName: efs-sc
+    storageClassDeploy: true
+#   storageClassName: efs-sc
```
Comment (ac74475) [inline values.yaml]: Is the storageClassName still used? If not can we delete this line

File: create-kafka-queues/src/main/resources/values.yaml [hunk 9]
```diff
@@ -17,6 +17,26 @@ replicaCount: 1
...
+    replicationFactor: 1
+    partitions: 1
+  - name: query-scope
+    replicationFactor: 1
+    partitions: 1
+  - name: results-controller
+    replicationFactor: 1
+    partitions: 1
+
 image:
```
Comment (ac74475) [inline create-kafka-queues/src/main/resources/values.yaml:40]: missing codeRelease

**Review by ac74475 — CHANGES_REQUESTED** [review]

File: create-kafka-queues/src/main/resources/values.yaml (hunk 9 above)
Comment (dev-958) [reply create-kafka-queues/src/main/resources/values.yaml:40]: it's not a service artifact - doesn't require it

File: values.yaml (hunk 8 above)
Comment (dev-958) [reply values.yaml]: no - I'll remove it

**Review by ac74475 — APPROVED** [review]
